### PR TITLE
Added docker support

### PR DESCRIPTION
**Pull request by SamantaTarun**
Please read the  docker_instructions.md file to build and run.

tarun@tarun-HP-Notebook:~/zestream-server$ docker image ls
REPOSITORY   TAG       IMAGE ID       CREATED             SIZE
go-dock      latest    f0d280ba3b02   13 minutes ago      558MB
<none>       <none>    6d4ff6536fbb   About an hour ago   992MB
golang       latest    6650307efe4a   5 days ago          992MB
golang       alpine    cae57157ceaa   2 weeks ago         354MB

Note: please install docker globally before using docker command.

#6

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    GOARCH=amd64
+
+# Move to working directory /build
+WORKDIR /build
+
+# Copy and download dependency using go mod
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
+
```
Comment (abskrj) [inline Dockerfile:16]: will also need to install ffmpeg with libx264 enabled while generating build

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
+FROM golang:alpine
```
Comment (abskrj) [inline Dockerfile:1]: also, do a little research on which distro will be based for video processing and supports GPU acceleration.

**Review by abskrj — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (KedarisettiSreeVamsi) [reply Dockerfile:1]: I found these resources for GPU accelerated containers:
https://learn.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/gpu-acceleration#run-a-container-with-gpu-acceleration
https://unrealcontainers.com/docs/concepts/gpu-acceleration#nvidia-gpu-support-for-linux-containers

Comment (KedarisettiSreeVamsi) [reply Dockerfile:1]: https://github.com/akofman/docker-video-processing

**Comment by abskrj** [comment]
@KedarisettiSreeVamsi  you taking this up?

**Comment by abskrj** [comment]
Closing this for now

**Pull request closed without merging**